src/api/customer/customer-controller.ts

```diff
@@ -1,20 +1,64 @@
-import { CREATED } from "http-status";
-import { persistCustomer } from "./customer-business";
+import { CREATED, NOT_FOUND, OK } from "http-status";
+import {
+  getAllCustomersBusiness,
+  getCustomerById,
+  persistCustomer,
+} from "./customer-business";
 import { createCustomerDeserializer } from "./customer-deserializer";
-import { createCustomerSerializer } from "./customer-serializer";
-import { CustomerCreationRequestHandler } from "./customer-type";
-import { createCustomerValidator } from "./customer-validator";
+import {
+  createCustomerSerializer,
+  getAllCustomersSerializer,
+  getCustomerSerializer,
+  paginationSerializer,
+} from "./customer-serializer";
+import {
+  CustomerCreationRequestHandler,
+  GetAllCustomerRequestHandler,
+  GetCustomerRequestHandler,
+} from "./customer-type";
+import {
+  createCustomerValidator,
+  getAllCustomersValidator,
+  getCustomerValidator,
+} from "./customer-validator";
 
 const createCustomer = (): CustomerCreationRequestHandler[] => {
-    return [
-        createCustomerValidator(),
-        createCustomerDeserializer,
-        persistCustomer,
-        createCustomerSerializer,
-        (req, res) => { res.status(CREATED).json(res.locals.customerToRespond) }
-    ]
-}
+  return [
+    createCustomerValidator(),
+    createCustomerDeserializer,
+    persistCustomer,
+    createCustomerSerializer,
+    (req, res) => {
+      res.status(CREATED).json(res.locals.customerToRespond);
+    },
+  ];
+};
+
+const getAllCustomers = (): GetAllCustomerRequestHandler[] => {
+  return [
+    getAllCustomersValidator(),
+    paginationSerializer,
+    getAllCustomersBusiness,
+    getAllCustomersSerializer,
+    (req, res) => {
+      res.status(OK).json(res.locals.customersToRespond);
+    },
+  ];
+};
 
-export {
-    createCustomer
+const getCustomersById = (): GetCustomerRequestHandler[] => {
+  return [
+    getCustomerValidator(),
+    getCustomerById,
+    getCustomerSerializer,
+    (req, res) => {
+      if (!res.locals.customerToRespond) {
+        res.status(NOT_FOUND).json(res.locals.customerToRespond);
+      } else {
+        res.status(OK).json(res.locals.customerToRespond);
+      }
```

Comment:
Seria melhor remover a negação `!` da expressão, e inverter o `if`.

```suggestion
      if (res.locals.customerToRespond) {
        res.status(OK).json(res.locals.customerToRespond);
      } else {
        res.status(NOT_FOUND).json(res.locals.customerToRespond);
      }
```